kv/txn_coord_sender.go

```diff
@@ -336,7 +336,9 @@ func (tc *TxnCoordSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*r
 					EndKey: endKey,
 				})
 			})
-			roachpb.MergeSpans(&et.IntentSpans)
+			// TODO(peter): Populate DistinctSpans on all batches, not just batches
+			// which contain an EndTransactionRequest.
+			ba.Header.DistinctSpans = roachpb.MergeSpans(&et.IntentSpans)
```

Comment:
Hmm, I bet the code is incorrect if there has been prior deduping of intent spans. I can change this so that it only sets `DistinctSpans` if `txnMeta.keys` was empty.